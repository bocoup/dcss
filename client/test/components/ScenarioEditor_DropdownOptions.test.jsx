import React from 'react';
import assert from 'assert';
import {
  fetchImplementation,
  mounter,
  reduxer,
  snapshotter,
  state,
} from '../bootstrap';
import { unmountComponentAtNode } from 'react-dom';
import { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  OwnerDropdown,
  CategoriesDropdown,
} from '../../components/ScenarioEditor/DropdownOptions.jsx';

const original = JSON.parse(JSON.stringify(state));
let container = null;
let commonProps = null;
let commonState = null;

beforeAll(() => {
  (window || global).fetch = jest.fn();
});

afterAll(() => {
  fetch.mockRestore();
});

beforeEach(() => {
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);

  commonProps = {
    history: {
      push() {},
    },
  };

  commonState = JSON.parse(JSON.stringify(original));
});

afterEach(() => {
  fetch.mockReset();
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  commonProps = null;
  commonState = null;
});

test('OwnerDropdown', () => {
  expect(OwnerDropdown).toBeDefined();
});

test('Snapshot 1 1', async (done) => {
  const Component = OwnerDropdown;

  const props = {
    ...commonProps,
    options: [],
    categories: [],
    author: { id: null },
    onChange() {},
  };

  const state = {
    ...commonState,
  };

  const reduxed = reduxer(Component, props, state);
  const wrapper = mounter(reduxed);
  expect(snapshotter(reduxed)).toMatchSnapshot();
  expect(snapshotter(wrapper)).toMatchSnapshot();

  const component = wrapper.findWhere((n) => {
    return n.type() === Component;
  });
  expect(snapshotter(component)).toMatchSnapshot();

  done();
});

test('CategoriesDropdown', () => {
  expect(CategoriesDropdown).toBeDefined();
});

test('Snapshot 1 1', async (done) => {
  const Component = CategoriesDropdown;

  const props = {
    ...commonProps,
    options: [],
    categories: [],
    author: { id: null },
    onChange() {},
  };

  const state = {
    ...commonState,
  };

  const reduxed = reduxer(Component, props, state);
  const wrapper = mounter(reduxed);
  expect(snapshotter(reduxed)).toMatchSnapshot();
  expect(snapshotter(wrapper)).toMatchSnapshot();

  const component = wrapper.findWhere((n) => {
    return n.type() === Component;
  });
  expect(snapshotter(component)).toMatchSnapshot();

  done();
});

/*{INJECTION}*/

