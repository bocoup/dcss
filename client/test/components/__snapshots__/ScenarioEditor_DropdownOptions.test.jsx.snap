// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1 1 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div
      class=\\"field\\"
    >
      <label>
        Owner
      </label>
      <div
        aria-expanded=\\"false\\"
        aria-multiselectable=\\"false\\"
        class=\\"ui fluid selection dropdown\\"
        label=\\"Owner\\"
        name=\\"author\\"
        role=\\"listbox\\"
        tabindex=\\"0\\"
      >
        <div
          aria-atomic=\\"true\\"
          aria-live=\\"polite\\"
          class=\\"default text\\"
          role=\\"alert\\"
        >
          Select...
        </div>
        <i
          aria-hidden=\\"true\\"
          class=\\"dropdown icon\\"
        />
        <div
          class=\\"menu transition\\"
        />
      </div>
    </div>
  </div>
</body>"
`;

exports[`Render 1 1 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div
      class=\\"field\\"
    >
      <label>
        Categories
      </label>
      <div
        aria-expanded=\\"false\\"
        aria-multiselectable=\\"true\\"
        class=\\"ui fluid multiple selection dropdown\\"
        label=\\"Categories\\"
        name=\\"categories\\"
        role=\\"listbox\\"
        tabindex=\\"0\\"
      >
        <div
          aria-atomic=\\"true\\"
          aria-live=\\"polite\\"
          class=\\"default text\\"
          role=\\"alert\\"
        >
          Select...
        </div>
        <i
          aria-hidden=\\"true\\"
          class=\\"dropdown icon\\"
        />
        <div
          class=\\"menu transition\\"
        />
      </div>
    </div>
  </div>
</body>"
`;
